kuksa_databroker/databroker/src/vss.rs

```diff
@@ -49,294 +151,325 @@ impl fmt::Display for Error {
+            types::DataType::Int64 => serde_json::from_value::<i64>(value)
+                .map(|value| Some(types::DataValue::Int64(value)))
+                .map_err(|err| err.into()),
+            types::DataType::Uint8 => serde_json::from_value::<u8>(value)
+                .map(|value| Some(types::DataValue::Uint32(value.into())))
+                .map_err(|err| err.into()),
+            types::DataType::Uint16 => serde_json::from_value::<u16>(value)
+                .map(|value| Some(types::DataValue::Uint32(value.into())))
+                .map_err(|err| err.into()),
+            types::DataType::Uint32 => serde_json::from_value::<u32>(value)
+                .map(|value| Some(types::DataValue::Uint32(value)))
+                .map_err(|err| err.into()),
+            types::DataType::Uint64 => serde_json::from_value::<u64>(value)
+                .map(|value| Some(types::DataValue::Uint64(value)))
+                .map_err(|err| err.into()),
+            types::DataType::Float => serde_json::from_value::<f32>(value)
+                .map(|value| Some(types::DataValue::Float(value)))
+                .map_err(|err| err.into()),
+            types::DataType::Double => serde_json::from_value::<f64>(value)
+                .map(|value| Some(types::DataValue::Double(value)))
+                .map_err(|err| err.into()),
+            types::DataType::Timestamp => {
+                Err(Error::ParseError("timestamp is not supported".to_owned()))
             }
-            // types::DataType::TimestampArray => todo!(),
-            _ => Err(Error::ParseError(format!(
-                "Could not parse default value ({}) as {:?} for {} (NOT IMPLEMENTED)",
-                value, data_type, name
-            ))),
+            types::DataType::StringArray => serde_json::from_value::<Vec<String>>(value)
+                .map(|array| Some(types::DataValue::StringArray(array)))
+                .map_err(|err| err.into()),
+            types::DataType::BoolArray => serde_json::from_value::<Vec<bool>>(value)
+                .map(|array| Some(types::DataValue::BoolArray(array)))
+                .map_err(|err| err.into()),
+            types::DataType::Int8Array => serde_json::from_value::<Vec<i8>>(value)
+                .map(|array| {
+                    Some(types::DataValue::Int32Array(
+                        array.iter().map(|value| (*value).into()).collect(),
+                    ))
+                })
+                .map_err(|err| err.into()),
+            types::DataType::Int16Array => serde_json::from_value::<Vec<i16>>(value)
+                .map(|array| {
+                    Some(types::DataValue::Int32Array(
+                        array.iter().map(|value| (*value).into()).collect(),
+                    ))
+                })
+                .map_err(|err| err.into()),
+            types::DataType::Int32Array => serde_json::from_value::<Vec<i32>>(value)
+                .map(|value| Some(types::DataValue::Int32Array(value)))
```

Comment:
How come it says `| value|` here and on line 314? Error or something i do not understand?

Reply:
That was pretty eagle-eyed of you :)

It's not an error, since it's only describes what to name the input variable in the closure for the `map` function.

It _is_ inconsistent, though, so I'll change it.